spanner/spanner_samples.rb

```diff
@@ -0,0 +1,390 @@
+# Copyright 2017 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def create_database project_id:, instance_id:, database_id:
+  # [START create_database]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  job = instance.create_database database_id, statements: [
+    "CREATE TABLE Singers (
+      SingerId     INT64 NOT NULL,
+      FirstName    STRING(1024),
+      LastName     STRING(1024),
+      SingerInfo   BYTES(MAX)
+    ) PRIMARY KEY (SingerId)",
+
+    "CREATE TABLE Albums (
+      SingerId     INT64 NOT NULL,
+      AlbumId      INT64 NOT NULL,
+      AlbumTitle   STRING(MAX)
+    ) PRIMARY KEY (SingerId, AlbumId),
+    INTERLEAVE IN PARENT Singers ON DELETE CASCADE"
+  ]
+
+  puts "Waiting for create database operation to complete"
+
+  job.wait_until_done!
+
+  puts "Created database #{database_id} on instance #{instance_id}"
+  # [END create_database]
+end
+
+def insert_data project_id:, instance_id:, database_id:
+  # [START insert_data]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  client.commit do |c|
+    c.insert "Singers", [
+      { SingerId: 1, FirstName: "Marc",     LastName: "Richards" },
```

Comment:
You aligned `{ }` in other locations. Should this be consistent here? Maybe not because it would leave a lot of whitespace.